yawl/lib/yawl/widget/Histogram.lua

```diff
@@ -0,0 +1,146 @@
+local class = require("libClass2")
+local Frame = require("yawl.widget.Frame")
+local gpu = require("component").gpu
+
+--[[
+    Ideally, Histogram will have horizontal scrolling. Though there is some contention about the direction.
+]]
+---@class Histogram:Frame
+---@field parent Frame
+---@operator call:Histogram
+---@overload fun(parent:Frame,x:number,y:number)
+local Histogram = class(Frame)
+
+---Comment
+---@return Histogram
+---@param parent Frame
+---@param x number
+---@param y number
+function Histogram:new(parent, x, y, maxColumns) --Histogramset can be string too
+    checkArg(1, parent, "table")
+    checkArg(1, maxColumns, "number", "nil")
+    local o = self.parent(parent, x, y)
+    setmetatable(o, {__index = self})
+    o._maxColumns = maxColumns
+    o._data = {}
+    o:fillChar(" ")
+    ---@cast o Histogram
+    return o
+end
+
+function Histogram:insert(value) --insert at the end
+    checkArg(1, value, "number")
+    table.insert(self._data, value)
+end
+
+function Histogram:set(index, value) --overwride the value
```

Comment:
this feel like a duplicate of adjust

Reply:
it's not. it's a getter/setter, adjust is an incrementer.